sqlite/src/printf.c

```diff
@@ -893,7 +893,7 @@ static int sqlite3StrAccumEnlarge(StrAccum *p, int N){
     char *zOld = isMalloced(p) ? p->zText : 0;
     i64 szNew = p->nChar;
     szNew += N + 1;
-    if( szNew+p->nChar<=p->mxAlloc ){
+    if( (p->noExtraAlloc==0) && (szNew+p->nChar<=p->mxAlloc) ){
       /* Force exponential buffer size growth as long as it does not overflow,
```

Comment:
Can we wrap this customization in an #ifdef block to prevent merge conflicts later?

Reply:
done